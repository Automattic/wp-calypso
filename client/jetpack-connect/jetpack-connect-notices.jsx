/** @format */
/**
 * External dependencies
 */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { localize } from 'i18n-calypso';

/**
 * Internal dependencies
 */
import Notice from 'components/notice';

class JetpackConnectNotices extends Component {
	static propTypes = {
		noticeType: PropTypes.oneOf( [
			'alreadyConnected',
			'alreadyConnectedByOtherUser',
			'alreadyOwned',
			'defaultAuthorizeError',
			'isDotCom',
			'jetpackIsValid',
			'notActiveJetpack',

			// notConnectedJetpack is expected, but no notice is shown.
			'notConnectedJetpack',
			'notExists',
			'notJetpack',
			'notWordPress',
			'outdatedJetpack',
			'retryAuth',
			'retryingAuth',
			'secretExpired',
			'wordpress.com',
		] ).isRequired,
		translate: PropTypes.func.isRequired,
		url: PropTypes.string,
	};

	getNoticeValues() {
		const { noticeType, onDismissClick, translate } = this.props;

		const noticeValues = {
			icon: 'notice',
			status: 'is-error',
			text: translate( "That's not a valid url." ),
			showDismiss: false,
		};

		if ( onDismissClick ) {
			noticeValues.onDismissClick = onDismissClick;
			noticeValues.showDismiss = true;
		}

		switch ( noticeType ) {
			case 'notExists':
				return noticeValues;

			case 'isDotCom':
				noticeValues.icon = 'block';
				noticeValues.text = translate(
					"That's a WordPress.com site, so you don't need to connect it."
				);
				return noticeValues;

			case 'notWordPress':
				noticeValues.icon = 'block';
				noticeValues.text = translate( "That's not a WordPress site." );
				return noticeValues;

			case 'notActiveJetpack':
				noticeValues.icon = 'block';
				noticeValues.text = translate( 'Jetpack is deactivated.' );
				return noticeValues;

			case 'outdatedJetpack':
				noticeValues.icon = 'block';
				noticeValues.text = translate( 'You must update Jetpack before connecting.' );
				return noticeValues;

			case 'jetpackIsDisconnected':
				noticeValues.icon = 'link-break';
				noticeValues.text = translate( 'Jetpack is currently disconnected.' );
				return noticeValues;

			case 'jetpackIsValid':
				noticeValues.status = 'is-success';
				noticeValues.icon = 'plugins';
				noticeValues.text = translate( 'Jetpack is connected.' );
				return noticeValues;

			case 'notJetpack':
				noticeValues.status = 'is-notice';
				noticeValues.icon = 'status';
				noticeValues.text = translate( "Jetpack couldn't be found." );
				return noticeValues;

			case 'wordpress.com':
				noticeValues.text = translate( "Oops, that's us." );
				noticeValues.status = 'is-warning';
				noticeValues.icon = 'status';
				return noticeValues;

			case 'retryingAuth':
				noticeValues.text = translate(
					'Error authorizing. Page is refreshing for another attempt.'
				);
				noticeValues.status = 'is-warning';
				noticeValues.icon = 'notice';
				return noticeValues;

			case 'retryAuth':
				noticeValues.text = translate(
					'In some cases, authorization can take a few attempts. Please try again.'
				);
				noticeValues.status = 'is-warning';
				noticeValues.icon = 'notice';
				return noticeValues;

			case 'secretExpired':
				noticeValues.text = translate( "Oops, that took a while. You'll have to try again." );
				noticeValues.status = 'is-error';
				noticeValues.icon = 'notice';
				return noticeValues;

			case 'defaultAuthorizeError':
				noticeValues.text = translate(
					'Error authorizing your site. Please {{link}}contact support{{/link}}.',
					{
						components: {
							link: (
								<a
									href="https://jetpack.com/contact-support"
									target="_blank"
									rel="noopener noreferrer"
								/>
							),
						},
					}
				);
				noticeValues.status = 'is-error';
				noticeValues.icon = 'notice';
				return noticeValues;

			case 'alreadyConnectedByOtherUser':
				noticeValues.text = translate(
					'This site is already connected to a different WordPress.com user, ' +
						'you need to disconnect that user before you can connect another.'
				);
				noticeValues.status = 'is-warning';
				noticeValues.icon = 'notice';
				return noticeValues;
			case 'userIsAlreadyConnectedToSite':
				noticeValues.text = translate(
					'This WordPress.com account is already connected to another user on this site. ' +
						'Please login to another WordPress.com account to complete the connection.'
				);
				noticeValues.status = 'is-warning';
				noticeValues.icon = 'notice';
				return noticeValues;
		}
	}

	render() {
		const values = this.getNoticeValues();
		if ( values ) {
			return (
				<div className="jetpack-connect__notices-container">
					<Notice { ...values } />
				</div>
			);
		}
		return null;
	}
}

export default localize( JetpackConnectNotices );
