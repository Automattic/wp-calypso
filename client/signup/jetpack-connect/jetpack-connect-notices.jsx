/**
 * External dependencies
 */
import React, { PropTypes } from 'react';

/**
 * Internal dependencies
 */
import Notice from 'components/notice';

export default React.createClass( {
	displayName: 'JetpackConnectNotices',

	propTypes: {
		noticeType: PropTypes.string,
		siteUrl: PropTypes.string
	},

	getNoticeValues( url ) {
		let noticeValues = {
			icon: 'trash',
			status: 'is-warning',
			text: this.translate( 'That\'s not a real web site' ),
			showDismiss: false
		};

		if ( this.props.onDismissClick ) {
			noticeValues.onDismissClick = this.props.onDismissClick;
			noticeValues.showDismiss = true;
		}

		if ( this.props.noticeType === 'notExists' ) {
			return noticeValues;
		}
		if ( this.props.noticeType === 'isDotCom' ) {
			noticeValues.icon = 'block';
			noticeValues.text = this.translate( 'That\'s a WordPress.com site, so you don\'t need to connect it' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'notWordPress' ) {
			noticeValues.icon = 'block';
			noticeValues.text = this.translate( 'That\'s not a WordPress site' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'notActiveJetpack' ) {
			noticeValues.icon = 'block';
			noticeValues.text = this.translate( 'Jetpack is deactivated' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'outdatedJetpack' ) {
			noticeValues.icon = 'block';
			noticeValues.text = this.translate( 'You need to update Jetpack before connecting' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'jetpackIsDisconnected' ) {
			noticeValues.icon = 'link-break';
			noticeValues.text = this.translate( 'Jetpack is disconnected' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'jetpackIsValid' ) {
			noticeValues.status = 'is-success';
			noticeValues.icon = 'plugins';
			noticeValues.text = this.translate( 'Jetpack is connected' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'notJetpack' ) {
			noticeValues.status = 'is-noticeType';
			noticeValues.icon = 'status';
			noticeValues.text = this.translate( 'Can\'t find Jetpack' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'alreadyConnected' ) {
			noticeValues.status = 'is-success';
			noticeValues.icon = 'status';
			noticeValues.text = this.translate( 'This site is already connected!' );
			return noticeValues;
		}
		if ( this.props.noticeType === 'alreadyOwned' ) {
			noticeValues.status = 'is-success';
			noticeValues.icon = 'status';
			noticeValues.text = this.translate( 'You already have connected {{a}}this site!{{/a}}', {
				components: {
					a: <a href={ '/stats/day/' + url } />
				}
			} );
			return noticeValues;
		}
		if ( this.props.noticeType === 'wordpress.com' ) {
			noticeValues.text = this.translate( 'I think that\'s us ¯\\_(ツ)_/¯' );
			noticeValues.status = 'is-warning';
			noticeValues.icon = 'status';
			return noticeValues;
		}
		if ( this.props.noticeType === 'secretExpired' ) {
			noticeValues.text = this.translate( 'Oops, that took a while. You\'ll have to try again.' );
			noticeValues.status = 'is-error';
			noticeValues.icon = 'notice';
			return noticeValues;
		}
		if ( this.props.noticeType === 'authorizeError' ) {
			noticeValues.text = this.translate( 'Error authorizing your site. Please contact support.' );
			noticeValues.status = 'is-error';
			noticeValues.icon = 'notice';
			return noticeValues;
		}
		return;
	},

	render() {
		const urlSlug = this.props.url.replace( /^https?:\/\//, '' ).replace( /\//g, '::' );
		const values = this.getNoticeValues( urlSlug );
		if ( values ) {
			return (
				<div className="jetpack-connect__notices-container">
					<Notice { ...values } />
				</div>
			);
		}
		return null;
	}
} );
