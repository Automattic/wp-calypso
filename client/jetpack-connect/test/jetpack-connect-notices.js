/**
 * @format
 * @jest-environment jsdom
 */
/**
 * External dependencies
 */
import React from 'react';
import { shallow } from 'enzyme';
import { identity } from 'lodash';

/**
 * Internal dependencies
 */
import { JetpackConnectNotices } from '../jetpack-connect-notices';

const terminalErrorNoticeType = 'notExists';
const nonTerminalErrorNoticeType = 'retryAuth';
const requiredProps = { translate: identity };

describe( 'JetpackConnectNotices', () => {
	test( 'Should render notice', () => {
		const wrapper = shallow(
			<JetpackConnectNotices { ...requiredProps } noticeType={ terminalErrorNoticeType } />
		);
		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.isEmptyRender() ).toBe( false );
	} );

	test( 'Should not render terminal notice if callback supplied', () => {
		const onTerminalError = jest.fn();
		const wrapper = shallow(
			<JetpackConnectNotices
				{ ...requiredProps }
				noticeType={ terminalErrorNoticeType }
				onTerminalError={ onTerminalError }
			/>
		);
		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.isEmptyRender() ).toBe( true );
	} );

	test( 'Should call callback on terminal error', () => {
		const onTerminalError = jest.fn();
		shallow(
			<JetpackConnectNotices
				{ ...requiredProps }
				noticeType={ terminalErrorNoticeType }
				onTerminalError={ onTerminalError }
			/>
		);
		expect( onTerminalError ).toHaveBeenCalledTimes( 1 );
	} );

	test( 'Should render non-terminal notice if callback supplied', () => {
		const onTerminalError = jest.fn();
		const wrapper = shallow(
			<JetpackConnectNotices
				{ ...requiredProps }
				noticeType={ nonTerminalErrorNoticeType }
				onTerminalError={ onTerminalError }
			/>
		);
		expect( onTerminalError ).not.toHaveBeenCalled();
		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.isEmptyRender() ).toBe( false );
	} );
} );
