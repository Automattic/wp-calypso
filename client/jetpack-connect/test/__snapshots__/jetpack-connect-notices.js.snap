// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JetpackConnectNotices Should not render terminal notice if callback supplied 1`] = `""`;

exports[`JetpackConnectNotices Should render non-terminal notice if callback supplied 1`] = `
<div
  className="jetpack-connect__notices-container"
>
  <Localized(Notice)
    icon="notice"
    showDismiss={false}
    status="is-warning"
    text="In some cases, authorization can take a few attempts. Please try again."
    userCanRetry={true}
  />
</div>
`;

exports[`JetpackConnectNotices Should render notice 1`] = `
<div
  className="jetpack-connect__notices-container"
>
  <Localized(Notice)
    icon="notice"
    showDismiss={false}
    status="is-error"
    text="That's not a valid url."
  />
</div>
`;
